src/sqlfluff/dialects/dialect_tsql.py

```diff
@@ -1236,6 +1236,7 @@ class DatatypeSegment(BaseSegment):
         Bracketed(
             OneOf(
                 Delimited(Ref("ExpressionSegment")),
+                "MAX",
```

Comment:
In the YAML below you can see the `MAX` is being parsed as a column identifier. Switch these around to give the `MAX` the precendence.

```suggestion
                "MAX",
                Delimited(Ref("ExpressionSegment")),
```

This will also allow other datatypes to use this (e.g. `VARCHAR(MAX)`). Not sure how much of a problem that would really be and if this simple option is fine? A more complex, but potentially more accurate solution is given here: https://github.com/sqlfluff/sqlfluff/issues/2710#issuecomment-1053556317

Reply:
I decided on this solution on purpose, as `varchar(max)` is also valid SQL 🙂

Reply:
OK. Did you see my comment about switching the order? Need to do that and regenerate the YAMLs as currently they are wrong.